src/create-code/lambda/identity.ts

```diff
@@ -0,0 +1,30 @@
+import fetch from "node-fetch";
+import {Response as FetchResponse} from "node-fetch";
+import {isProd} from "../lib/stage";
+
+const idapiBaseUrl = isProd() ? "https://idapi.theguardian.com" : "https://idapi.code.dev-theguardian.com";
+
+export function getBrazeUuidByEmail(email: string, accessToken: string): Promise<string> {
+    return fetch(
+        `${idapiBaseUrl}/user?emailAddress=${email}`,
+        {headers: {"X-GU-ID-Client-Access-Token": `Bearer ${accessToken}`}}
+        ).then((response: FetchResponse) => {
+            if (response.status == 404) {
+                return Promise.reject(new Error(`Email not found: ${email}`));
+            }
+            if (!response.ok) {
+                return Promise.reject(new Error(`Identity API user endpoint responded with status: ${response.status}`));
+            }
+            return response.json();
+        }).then((identityResponse) => {
+            if (
+                identityResponse.user &&
+                identityResponse.user.privateFields &&
+                identityResponse.user.privateFields.brazeUuid
```

Comment:
I did try updating the TS version so that I can use `identityResponse?.user?.privateFields?.brazeUuid` but it caused other issues with the build, so I don't to lose time on this now